go-controller/pkg/ovn/egressip.go

```diff
@@ -327,6 +327,12 @@ func (oc *DefaultNetworkController) reconcileEgressIPPod(old, new *v1.Pod) (err
 		oldPod = old
 		namespace, err = oc.watchFactory.GetNamespace(oldPod.Namespace)
 		if err != nil {
+			// when the whole namespace gets removed, we can ignore the NotFound error here
+			// any potential configuration will get removed in reconcileEgressIPNamespace
+			if new == nil && apierrors.IsNotFound(err) {
```

Comment:
`reconcileEgressIPPod` doesn't fetch a pod so it won't be directly affected by pod not existing anymore.
As for namespace not found I think we should still return an error on update since this seems like a messed up setup when there are pods running in non-existing namespaces.